frontend/javascripts/admin/time/time_line_view.tsx

```diff
@@ -89,30 +101,94 @@ class TimeLineView extends React.PureComponent<Props, State> {
     },
     isLoading: false,
     isFetchingUsers: false,
+    initialUserId: null,
+    selectedProjectIds: [],
+    allProjects: [],
+    annotationType: AnnotationTypeFilterEnum.TASKS_AND_ANNOTATIONS_KEY,
   };
 
-  componentDidMount() {
-    const isAdminOrTeamManger = isUserAdminOrTeamManager(this.props.activeUser);
+  parseQueryParams() {
+    const params = Utils.getUrlParamsObject();
+    let dateRange: DateRange = [dayjs().startOf("day"), dayjs().endOf("day")];
+    if (params == null) return { initialUserId: null };
+    const hasStart = _.has(params, "start");
+    const hasEnd = _.has(params, "end");
+    // if either start or end is provided, use one week as default range
+    if (!hasStart && hasEnd) {
+      const end = dayjs(+params.end); // + leads to the timestamp being parsed as number; omitting it leads to wrong parsing
+      dateRange = [end.subtract(7, "d"), end];
+    } else if (hasStart && !hasEnd) {
+      const start = dayjs(+params.start);
+      dateRange = [start, start.add(7, "d")];
+    } else if (hasStart && hasEnd) {
+      dateRange = [dayjs(+params.start), dayjs(+params.end)];
+    }
+    let selectedProjectIds: Array<string> = [];
+    let annotationType = AnnotationTypeFilterEnum.TASKS_AND_ANNOTATIONS_KEY;
+    if (_.has(params, "projectsOrType")) {
+      const projectsOrTypeParam = params.projectsOrType;
+      if (Object.values<string>(AnnotationTypeFilterEnum).includes(projectsOrTypeParam)) {
+        annotationType = projectsOrTypeParam as AnnotationTypeFilterEnum;
+      } else {
+        selectedProjectIds = params.projectsOrType.split(",");
+      }
+    }
+    this.setState(
+      {
+        initialUserId: _.has(params, "user") ? params.user : null,
+        dateRange,
+        annotationType,
+        selectedProjectIds,
+      },
+      () => {
+        this.handleDateChange(dateRange);
+      },
+    );
+  }
 
+  async componentDidMount() {
+    const isAdminOrTeamManger = isUserAdminOrTeamManager(this.props.activeUser);
+    this.parseQueryParams();
     if (isAdminOrTeamManger) {
-      this.fetchData();
+      await this.fetchData();
+      await this.fetchTimeTrackingData();
     } else {
       this.fetchDataFromLoggedInUser();
     }
   }
 
+  async componentDidUpdate(_prevProps: Props, prevState: State) {
+    if (
+      prevState.annotationType !== this.state.annotationType ||
+      prevState.selectedProjectIds !== this.state.selectedProjectIds
+    ) {
+      await this.fetchTimeTrackingData();
+    }
+  }
+
   async fetchData() {
     this.setState({
       isFetchingUsers: true,
     });
     const users = await getEditableUsers();
+    const currentUser =
+      this.state.initialUserId != null && isUserAdminOrTeamManager(this.props.activeUser)
+        ? await getUser(this.state.initialUserId).catch((err) => {
+            Toast.error("User not found.");
+            console.error(err.messages[0].error); // TODO is there a more stable way?
+            return null;
+          })
```

Comment:
right. I added this because I was running into this case a lot (with this organization-issue that florian has already fixed) and I didnt know what was going on at all. This shouldnt really happen all that often now, so I'll remove it :)